app/policies/map_policy.rb

```diff
@@ -4,7 +4,8 @@ def resolve
       visible = ['public', 'commons']
       permission = 'maps.permission IN (?)'
       if user
-        scope.where(permission + ' OR maps.user_id = ?', visible, user.id)
+        shared_maps = user.user_maps.map(&:id)
```

Comment:
Note to self &:id should be &:map_id in actuality I believe... Currently its a bug